src/agent.cc

```diff
@@ -151,6 +151,10 @@ void Agent::Build(Agent* parent) {
 
 void Agent::EnterNotify() {
   ctx_->RegisterAgent(this);
+
+  usagesdata.RecordMetadatas(this);
```

Comment:
"usages" is probably supposed to be "usage"
```suggestion
  usagedata.RecordMetadata(this);
```